### DISCUSSION TITLE
BindableState - Binding to state of different type

### DISCUSSION THREAD

**Discussion by nmccann** [pointfreeco/swift-composable-architecture#808] [Q&A] [answered]
Hi,

I've recently started migrating to BindableState, but in the process I've run into a situation that seems like it should have an easy answer, but I'm unable to figure it out.

I'm working on a login form with the following state (simplified):

```
struct LoginState: Equatable {
@BindableState var password: Redacted<String> = .init(secret: "")
}
```

The `Redacted` type is a wrapper that helps make it obvious that a given type shouldn't be included in any logging (using a custom mirror).

Later, in my view, I would like to create a binding. I can't use: `SecureField("Password", text: viewStore.$password)` because that would only work with a `String`. I was able to get around this with:

```
SecureField("Password", text: viewStore.binding(get: { $0.password.secret },
                                                        send: { .binding(.set(\.password, .init(secret: $0)))})
```

While the above works, it emits the following deprecation warning: `'set' is deprecated: For improved safety, bindable properties must now be wrapped explicitly in 'BindableState', and accessed via key paths to that 'BindableState', like '\.$value'`

What would be the recommended way to bind a value of one type to a different type when using `@BindableState`?

Thank you for any guidance you can provide!

Note: I have considered just using a String and making my `LoginState` conform to `CustomDumpReflectable` to omit the password, but that still leaves it in the action logs (so I'd likely have to make `LoginAction` conform to that as well, which further complicates things - having a single `Redacted` type would be preferable)

**Comment by mbrandonw** [comment, marked as answer]
If the `secret` field in `Redacted` is a mutable var, then you should be able to just do this:

```swift
SecureField("Password", text: viewStore.$password.secret)
```

**Comment by nmccann** [comment]
Yup - that worked. Thank you for the quick response!
